clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AvroComputeAggregationRequestBuilder.java

```diff
@@ -0,0 +1,131 @@
+  private final AvroComputeRequestBuilderV3<K> delegate;
+  private final Map<String, Integer> fieldTopKMap = new HashMap<>();
+  private final Map<String, Map<String, Predicate>> fieldBucketMap = new HashMap<>();
+  private final SchemaReader schemaReader;
+
+  public AvroComputeAggregationRequestBuilder(
+      AvroGenericReadComputeStoreClient storeClient,
+      SchemaReader schemaReader) {
+    this.delegate = (AvroComputeRequestBuilderV3<K>) storeClient.compute();
+    this.schemaReader = schemaReader;
+  }
+
+  @Override
+  public ComputeAggregationRequestBuilder<K> countGroupByValue(int topK, String... fieldNames) {
+    // topK must bigger than 0
+    if (topK <= 0) {
+      throw new VeniceClientException("TopK must be positive");
+    }
+    // field name must not be empty
+    if (fieldNames == null || fieldNames.length == 0) {
+      throw new VeniceClientException("fieldNames cannot be null or empty");
+    }
+
+    // Validate fields exist in schema
+    Schema valueSchema = schemaReader.getValueSchema(schemaReader.getLatestValueSchemaId());
+    for (String fieldName: fieldNames) {
+      if (fieldName == null) {
+        throw new VeniceClientException("Field name cannot be null");
+      }
+      if (fieldName.isEmpty()) {
+        throw new VeniceClientException("Field name cannot be empty");
+      }
+
+      Schema.Field field = valueSchema.getField(fieldName);
+      if (field == null) {
+        throw new VeniceClientException("Field not found in schema: " + fieldName);
+      }
+
+      // Store topK value for each field
+      fieldTopKMap.put(fieldName, topK);
+
+      // For countGroupByValue, we need to project the field itself
+      delegate.project(fieldName);
+    }
+    return this;
+  }
+
+  @Override
+  public <T> ComputeAggregationRequestBuilder<K> countGroupByBucket(
+      Map<String, Predicate<T>> bucketNameToPredicate,
+      String... fieldNames) {
+    // bucket predicates must not be null or empty
+    if (bucketNameToPredicate == null || bucketNameToPredicate.isEmpty()) {
+      throw new VeniceClientException("bucketNameToPredicate cannot be null or empty");
+    }
+    // field names must not be null or empty
+    if (fieldNames == null || fieldNames.length == 0) {
+      throw new VeniceClientException("fieldNames cannot be null or empty");
+    }
+
+    // Validate bucket names and predicates
+    for (Map.Entry<String, Predicate<T>> entry: bucketNameToPredicate.entrySet()) {
+      if (entry.getKey() == null || entry.getKey().isEmpty()) {
+        throw new VeniceClientException("Bucket name cannot be null or empty");
+      }
+      if (entry.getValue() == null) {
+        throw new VeniceClientException("Predicate for bucket '" + entry.getKey() + "' cannot be null");
+      }
+    }
+
+    // Validate fields exist in schema
+    Schema valueSchema = schemaReader.getValueSchema(schemaReader.getLatestValueSchemaId());
```

Comment:
Maybe we could define a private function for this validation logic, it is shared with countGroupByValue